src/Core/Util/Form.php

```diff
@@ -0,0 +1,444 @@
+                        },
+                        (array)Hash::extract($one, 'items.enum')
+                    );
+                }
+            }
+            if (!empty($options)) {
+                return [
+                    'type' => 'select',
+                    'options' => $options,
+                    'multiple' => 'checkbox',
+                ];
+            }
+        }
+
+        return [
+            'type' => 'checkbox',
+            'checked' => (bool)$value,
+        ];
+    }
+
+    /**
+     * Control for enum
+     *
+     * @param array $schema Object schema array.
+     * @param string $type Control type.
+     * @param mixed|null $value Property value.
+     * @return array
+     */
+    private static function enumControl(array $schema, string $type, $value) : array
+    {
+        if (!empty($schema['oneOf'])) {
+            foreach ($schema['oneOf'] as $one) {
+                if (!empty($one['enum'])) {
+                    $schema['enum'] = $one['enum'];
+                    array_unshift($schema['enum'], '');
+                }
+            }
+        }
+
+        return [
+            'type' => 'select',
+            'options' => array_map(
+                function ($value) {
+                    return ['value' => $value, 'text' => Inflector::humanize($value)];
+                },
+                $schema['enum']
+            ),
+        ];
+    }
+
+    /**
+     * Infer control type from property schema
+     * Possible return values:
+     *
+     *   'text'
+     *   'date-time'
+     *   'date'
+     *   'textarea'
+     *   'enum'
+     *   'number'
+     *   'checkbox'
+     *   'json'
+     *
+     * @param mixed $schema The property schema
+     * @return string
+     */
+    public static function controlTypeFromSchema($schema) : string
+    {
+        if (!is_array($schema)) {
+            return 'text';
+        }
+        if (!empty($schema['oneOf'])) {
+            foreach ($schema['oneOf'] as $subSchema) {
+                if (!empty($subSchema['type']) && $subSchema['type'] === 'null') {
+                    continue;
+                }
+
+                return Form::controlTypeFromSchema($subSchema);
+            }
+        }
+        if (empty($schema['type']) || !in_array($schema['type'], Form::SCHEMA_PROPERTY_TYPES)) {
+            return 'text';
+        }
+        $method = sprintf('typeFrom%s', lcfirst($schema['type']));
+
+        return Form::{$method}($schema);
+    }
+
+    /**
+     * Get controls option by property name.
+     *
+     * @param string $name The field name.
+     * @return array
+     */
+    public static function customControlOptions($name) : array
+    {
+        if (!in_array($name, Form::CUSTOM_CONTROLS)) {
+            return [];
+        }
+        $method = sprintf('%sOptions', $name);
+        $method = str_replace('-', '', $method);
+        $method = str_replace('_', '', $method);
```

Comment:
I would replace with
```php
$method = sprintf('%sOptions', Inflector::variable(str_replace('-', '_'. $name)));
```

`old_password` becomes `oldPasswordOptions` and `confirm-password` becomes `confirmPasswordOptions`